### DISCUSSION TITLE
Dagster DBTCLiResource initialisation without passing project directory path.

### DISCUSSION THREAD

**Discussion by zafarmah92** [dagster-io/dagster#21394] [Q&A] [answered]
Our DBT project is hosted somewhere else outside the dagster world. Is there a way we can use DBTCLiResource without passing/mentioning project_dir parameter. 

previously `(dagster-dbt <= 1.3) ` it was possible and was easy to override dbt cli methods. However, from and onwards `v1.4` there are lot of breaking changes with the  introduction of DbtCliResource and DbtCLiInvocation. 

at the moment, I have manifest.json file and can declare the asset like below. 

`# manifest.json is in target folder 
@dbt_assets(manifest=Path("target", "manifest.json"))
def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
    yield from dbt.cli(["build"], context=context).stream()`
    
  However I can't use this asset in Definitions (...) where the resource key 'dbt : DbtCliResource' excepts to be initialised with project directory.
  
 
 some suggestions would be helpful, do I have to write forked version of these resource classes ?

**Comment by rexledesma** [comment, marked as answer]
> Is there a way we can use DBTCLiResource without passing/mentioning project_dir parameter.

`DbtCliResource` requires the `project_dir` parameter because you cannot execute the `dbt` CLI without access to the underlying dbt project: https://docs.getdbt.com/reference/dbt_project.yml.

> Our DBT project is hosted somewhere else outside the dagster world.

If your dbt project is hosted in a different git repository from your Dagster code, we recommend that you copy the dbt project over to your Dagster project at build time: https://github.com/dagster-io/dagster/discussions/13767#discussioncomment-5664883.
